From e787362603b6d513a6e159b2a82e7faf56fdd88f Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Sat, 21 Mar 2020 20:42:50 +0200
Subject: [PATCH 1117/1239] tools/mrvl_uart: remove fast download option

The baudrate speed-up was a preparation for SoCs based on AP810 die.
Since AP810 has been phased out by Marvell, this support is not needed
anymore and should be removed to avoid customers confusion.

Change-Id: Ia646d5895325a9fdc9d23bd28b5b30918050059e
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/25307
Reviewed-by: Yan Markman <ymarkman@marvell.com>
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
---
 tools/mrvl_uart.sh | 42 ++++++------------------------------------
 1 file changed, 6 insertions(+), 36 deletions(-)

diff --git a/tools/mrvl_uart.sh b/tools/mrvl_uart.sh
index a46411fc99..e662faf745 100755
--- a/tools/mrvl_uart.sh
+++ b/tools/mrvl_uart.sh
@@ -8,7 +8,7 @@
 #
 # Author: Konstantin Porotchkin kostap@marvell.com
 #
-# Version 0.3
+# Version 0.4
 #
 # UART recovery downloader for Armada SoCs
 #
@@ -16,44 +16,21 @@
 
 port=$1
 file=$2
-speed=$3
 
 pattern_repeat=1500
 default_baudrate=115200
 tmpfile=/tmp/xmodem.pattern
 tools=( dd stty sx minicom )
 
-case "$3" in
-    2)
-        fast_baudrate=230400
-        prefix="\xF2"
-        ;;
-    4)
-        fast_baudrate=460800
-        prefix="\xF4"
-        ;;
-    8)
-    	fast_baudrate=921600
-        prefix="\xF8"
-        ;;
-    *)
-    	fast_baudrate=$default_baudrate
-        prefix="\xBB"
-esac
-
 if [[ -z "$port" || -z "$file" ]]
 then
     echo -e "\nMarvell recovery image downloader for Armada SoC family."
     echo -e "Command syntax:"
-    echo -e "\t$(basename $0) <port> <file> [2|4|8]"
+    echo -e "\t$(basename $0) <port> <file>"
     echo -e "\tport  - serial port the target board is connected to"
     echo -e "\tfile  - recovery boot image for target download"
-    echo -e "\t2|4|8 - times to increase the default serial port speed by"
-    echo -e "For example - load the image over ttyUSB0 @ 460800 baud:"
-    echo -e "$(basename $0) /dev/ttyUSB0 /tmp/flash-image.bin 4\n"
-    echo -e "=====WARNING====="
-    echo -e "- The speed-up option is not available in SoC families prior to A8K+"
-    echo -e "- This utility is not compatible with Armada 37xx SoC family\n"
+    echo -e "For example - load the image over ttyUSB0:"
+    echo -e "$(basename $0) /dev/ttyUSB0 /tmp/flash-image.bin\n"
 fi
 
 # Sanity checks
@@ -84,10 +61,6 @@ do
     fi
 done
 
-
-echo -e "Recovery will run at $fast_baudrate baud"
-echo -e "========================================"
-
 if [ -f "$tmpfile" ]
 then
     rm -f $tmpfile
@@ -97,7 +70,7 @@ fi
 stty -F $port raw ignbrk time 5 $default_baudrate
 counter=0
 while [ $counter -lt $pattern_repeat ]; do
-    echo -n -e "$prefix\x11\x22\x33\x44\x55\x66\x77" >> $tmpfile
+    echo -n -e "0xBB\x11\x22\x33\x44\x55\x66\x77" >> $tmpfile
     let counter=counter+1
 done
 
@@ -107,13 +80,10 @@ read
 dd if=$tmpfile of=$port &>/dev/null
 
 # Speed up the binary image transfer
-stty -F $port raw ignbrk time 5 $fast_baudrate
+stty -F $port raw ignbrk time 5 $default_baudrate
 sx -vv $file > $port < $port
 #sx-at91 $port $file
 
-# Return the port to the default speed
-stty -F $port raw ignbrk time 5 $default_baudrate
-
 # Optional - fire up Minicom
 minicom -D $port -b $default_baudrate
 
-- 
2.29.0

